src/gui/src/LogWindow.cpp

```diff
@@ -32,33 +34,58 @@ LogWindow::LogWindow(QWidget *parent) :
     // repeatedly until InputLeap is finished
     setAttribute(Qt::WA_DeleteOnClose, false);
     setupUi(this);
+    
+    // Use a timer to flush the buffer every 100 milliseconds
+    QTimer* timer = new QTimer(this);
+    connect(timer, &QTimer::timeout, this, &LogWindow::flushBuffer);
+    timer->start(100);
 }
 
 void LogWindow::startNewInstance()
 {
     // put a space between last log output and new instance.
-    if (!m_pLogOutput->toPlainText().isEmpty())
-        appendRaw("");
+    if (!buffer.isEmpty())
+        buffer += '\n';
 }
 
 void LogWindow::appendInfo(const QString& text)
 {
-    appendRaw(getTimeStamp() + " INFO: " + text);
+    buffer += getTimeStamp() + " INFO: " + text + '\n';
 }
 
 void LogWindow::appendDebug(const QString& text)
 {
-    appendRaw(getTimeStamp() + " DEBUG: " + text);
+    buffer += getTimeStamp() + " DEBUG: " + text + '\n';
 }
 
 void LogWindow::appendError(const QString& text)
 {
-    appendRaw(getTimeStamp() + " ERROR: " + text);
+    buffer += getTimeStamp() + " ERROR: " + text + '\n';
 }
 
 void LogWindow::appendRaw(const QString& text)
 {
-    m_pLogOutput->append(text);
+    buffer += text + '\n';
+}
+
+void LogWindow::flushBuffer()
+{
+    if (!buffer.isEmpty())
+    {
+        // Insert the new log content from the buffer
+        m_pLogOutput->appendPlainText(buffer);
+        buffer.clear();
+
+        // Scroll to the bottom if it was at the bottom before
```

Comment:
Pretty sure that you can set a property in the widget for this behavior

Reply:
@hunterzero99 I considered the review and made https://github.com/prahal/input-leap/tree/wip-log-window-fixes above input-leap/master.

I disabled the scroll-to-bottom code because it only stopped scrolling to the bottom if one clicked the text. Not if one clicked the scrollbar or scrolled with the mouse.
As far as I tested the output in the log window already scrolls with new content if I do not act upon it.

I also made the buffer private.

I did not code much in cpp so I took the liberty to make the "QString m_infoLine = QStringLiteral("%1 INFO: %2 \n");" into private attributes. Maybe there is a cleaner way.

I tested this code.
The only issue I have left is that with input-leap set to DEBUG2 mode and log to file the start of the log is not saved to file.
I believe the log-to-file is made from the LogWindow content and this LogWIndow content being truncated to 10000 lines cut the start of the logs (thus the version and other critical data).

Maybe the log-to-file should be handled by your new flushBuffer instead of proxying through the LogWindow code. That would solve this blocker.

Reply:
@sithlord48 about setting a property to get this behavior, did you mean "scroll to the bottom if it was at the bottom before"?
Note this behavior is the default one and this code "to scroll to bottom" introduces bugs (it scrolls to bottom even if we use the mouse to move in the QPlaintTextEdit. So I removed it altogether in https://github.com/prahal/input-leap/commit/2192f7233c4196692d7a90dd24fe68a865e43e0c

@hunterzero99 do you want me to take over with a PR from https://github.com/prahal/input-leap/commit/2192f7233c4196692d7a90dd24fe68a865e43e0c bringing @sithlord48 changes to your code?

I ran input-leap with max debug2 logs and had no hangs for days. My modification looks ready to go to master to me.

Reply:
That sounds great to me @prahal, thank you!  I haven't had a chance to build from your changes yet, but they seem good to me at a visual review.